spikeinterface/sortingcomponents/dimensionality_reduction.py

```diff
@@ -0,0 +1,99 @@
+import pickle
+
+import numpy as np
+
+from spikeinterface.sortingcomponents.peak_pipeline import run_peak_pipeline, PeakPipelineStep
+from spikeinterface.sortingcomponents.peak_detection import detect_peaks
+from spikeinterface.sortingcomponents.peak_selection import select_peaks
+from spikeinterface.postprocessing import compute_principal_components
+from spikeinterface.core.template_tools import get_template_channel_sparsity
+from spikeinterface import extract_waveforms
+from spikeinterface import NumpySorting
+
+
+class WaveformStep(PeakPipelineStep):
+    need_waveforms = True
+
+    def __init__(self, recording, ms_before=1., ms_after=1.,  peak_sign='neg', all_channels=True):
+        PeakPipelineStep.__init__(self, recording, ms_before=ms_before, ms_after=ms_after)
+        self.all_channels = all_channels
+        self.peak_sign = peak_sign
+        self._kwargs.update(dict(all_channels=all_channels, peak_sign=peak_sign))
+        self._dtype = recording.get_dtype()
+
+    def get_dtype(self):
+        return self._dtype
+
+    
+    def compute_buffer(self, traces, peaks, waveforms):
+        
+        return waveforms
+
+class TemporalPCA(PeakPipelineStep):
+    need_waveforms = True
+
+    def __init__(self, recording, ms_before=1., ms_after=1.,  peak_sign='neg', all_channels=True, model_path=None, local_radius_um=None):
+        PeakPipelineStep.__init__(self, recording, ms_before=ms_before, ms_after=ms_after, local_radius_um=local_radius_um)
+        self.all_channels = all_channels
+        self.peak_sign = peak_sign
+        self._kwargs.update(dict(all_channels=all_channels, peak_sign=peak_sign, model_path=model_path))
+        self._dtype = recording.get_dtype()
+
+    def get_dtype(self):
+        return self._dtype
+
+    def fit(self, recording, n_components, detect_peaks_params, peak_selection_params, whiten=True, **job_kwargs):
+        
+        # Detect peaks and sub-sample them
+        peaks = detect_peaks(recording, **detect_peaks_params, **job_kwargs)
+        peaks = select_peaks(peaks, **peak_selection_params) # How to select n_peaks
+
+        # Create a waveform extractor
+        ms_before = self._kwargs["ms_before"]
+        ms_after = self._kwargs["ms_after"]
+        sorting = NumpySorting.from_peaks(peaks, sampling_frequency=recording.sampling_frequency) 
+        we = extract_waveforms(recording, sorting, ms_before=ms_before, ms_after=ms_after, folder=None, 
+                               mode="memory", max_spikes_per_unit=None, **job_kwargs)
```

Comment:
With the new sparsity object (#1152 ), we could either already extract sparse wavefroms as:

```suggestion
        we = extract_waveforms(recording, sorting, ms_before=ms_before, ms_after=ms_after, folder=None, 
                               mode="memory", max_spikes_per_unit=None, sparse=True, method="radius", radius_um=self.local_radius_um, **job_kwargs)
```